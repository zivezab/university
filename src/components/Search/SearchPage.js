import React, { useState } from "react";
import SearchInput from "./SearchInput";
import UniversityCards from "./UniversityCards";

const SearchPage = (props) => {
  const [keyword, setKeyword] = useState("");
  const [country, setCountry] = useState("");
  const [favoriteList, setFavoriteList] = useState();
  const [universityList, setUniversityList] = useState();
  const [showError, setShowError] = useState(false);
  const [disableInput, setDisableInput] = useState(false);

  const exportJSONFile = (data) => {
    let dataStr = JSON.stringify(universityList);
    let dataUri =
      "data:application/json;charset=utf-8," + encodeURIComponent(dataStr);
    let exportFileDefaultName = "users.json";

    let linkElement = document.createElement("a");
    linkElement.setAttribute("href", dataUri);
    linkElement.setAttribute("download", exportFileDefaultName);
    linkElement.click();
  };

  const fetchData = async (e) => {
    e.preventDefault();
    setDisableInput(true);
    try {
      let url =
        "http://universities.hipolabs.com/search?country=" +
        country +
        "&name=" +
        keyword;
      await fetch(url)
        .then((response) => response.json())
        .then((data) => {
          setUniversityList(data);
        });
    } catch (error) {
      setShowError(true);
    }
    setDisableInput(false);
  };

  const onKeywordChange = async (keyword) => {
    setKeyword(keyword);
  };

  const onCountryChange = async (country) => {
    setCountry(country);
  };

  const removeFavorite = async (favorite) => {
    const favList = Array.isArray(favoriteList) ? [...favoriteList] : [];
    setFavoriteList(favList.filter((obj) => obj.name !== favorite.name));
  };

  const addFavorite = async (favorite) => {
    const favList = Array.isArray(favoriteList) ? [...favoriteList] : [];
    favList.push(favorite);
    setFavoriteList(favList);
  };

  return (
    <>
      <SearchInput
        onKeywordChange={onKeywordChange}
        onCountryChange={onCountryChange}
        onSubmit={fetchData}
        exportData={exportJSONFile}
        disabled={disableInput}
      />
      <UniversityCards
        addFavorite={addFavorite}
        removeFavorite={removeFavorite}
        universityList={universityList}
        showError={showError}
        favoriteList={favoriteList}
      />
    </>
  );
};

export default SearchPage;
