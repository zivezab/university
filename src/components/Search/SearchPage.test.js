import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";

import SearchPage from "./SearchPage";

test("matches snapshot", () => {
  const wrapper = shallow(<SearchPage />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});
