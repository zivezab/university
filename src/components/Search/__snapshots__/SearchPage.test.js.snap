// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <SearchInput
    disabled={false}
    exportData={[Function]}
    onCountryChange={[Function]}
    onKeywordChange={[Function]}
    onSubmit={[Function]}
  />
  <UniversityCards
    addFavorite={[Function]}
    favoriteList={Array []}
    removeFavorite={[Function]}
    showError={false}
    universityList={Array []}
  />
</Fragment>
`;
